gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -10,16 +10,32 @@ module T::Private::Methods
   ARG_NOT_PROVIDED = Object.new
   PROC_TYPE = Object.new
 
+  ALLOWED_SIG_ARGS = [].freeze
   DeclarationBlock = Struct.new(:mod, :loc, :blk)
 
-  def self.declare_sig(mod, &blk)
+  def self.declare_sig(mod, args, &blk)
     install_hooks(mod)
 
     if T::Private::DeclState.current.active_declaration
       T::Private::DeclState.current.reset!
       raise "You called sig twice without declaring a method inbetween"
     end
 
+    ALLOWED_SIG_ARGS.each do |allowed|
+      old_len = args.length
+      args.select! { |a| a != allowed }
+      diff = old_len - args.length
+      if diff == 0
+        next
+      end
+      if diff > 1
+        raise "Duplicate argument to `sig`: #{allowed}"
+      end
+    end
+    if args.length != 0
+      raise "Some arguments to `sig` were invalid: #{args}"
+    end
```

Comment:
Can we not over-generalize? It seems like this should be as easy as

```ruby
args.length == 0 || (args.length == 1 && arg[0] == :final)
```

Reply:
Similarly this means we wouldn't have to use `*args` in the sig for sig, we could just do `is_final = nil` to have a default arg.

Reply:
I was thinking doing it like this would make it easier if in the future, we want to move `:abstract` and `:overrides` and things like that to this part of the sig. Should I just not bother?

Reply:
Well first I think this overcomplicates the arg parsing logic even if there were multiple. Consider:

```ruby
is_final = false
is_abstract = false

args.each do |arg|
  case arg
  when :final
    is_final = true
  when :abstract
    is_abstract = true
  else
    raise "Unrecognized arg: #{arg}"
end
```

And **because** we don't have two or more dots to connect yet, I don't think we should generalize.

Reply:
e.g., we don't even know whether we want `sig(:final)` or `sig.final` or `sig {...}.final` or what. So let's write the least amount of code possible, so that it's as easy as possible to remove it.